### PR TITLE
Refactor - Simplify React Components

### PR DESCRIPTION
**Pull request by vogelino**
This PR goes about reducing complexity and renaming a bunch of different react components under the Card folder. Those were unnecessarily placed under the Card components, unused or badly named.

Remaining refactorings to do: (once #287 is merged):
- [ ] Remove the `CardDescription` component from `Legend` and use `SmallParagraph` or a custom `StyledComponent` instead
- [ ] Rename and restructure the `Card` component to be more explicit about its purpose. Eg. `TreeInformation`

### CODE REVIEW

File: src/components/Sidebar/SidebarSearch/index.tsx [hunk 1]
```diff
@@ -1,12 +1,88 @@
...
+
+const FilterLinksContainer = styled.div`
+  display: flex;
+  gap: 8px;
+  justify-content: flex-start;
+  flex-wrap: wrap;
+  padding-top: 16px;
+`;
+
+const FilterLink = styled.span<{ active?: boolean }>`
```
Comment (ff6347) [inline src/components/Sidebar/SidebarSearch/index.tsx:19]: prefer `isActive` ;)

File: src/components/Sidebar/SidebarSearch/index.tsx [hunk 2]
```diff
@@ -1,12 +1,88 @@
...
+        <SmallParagraph>
+          Betrachte welche Bäume bereits von anderen Nutzern gegossen wurden.
+          Oder finde heraus, wieviel Niederschlag die Bäume in den letzten 30
+          Tagen erreicht hat.
+        </SmallParagraph>
+        <FilterLinksContainer>
+          <FilterLink
+            active={dataView === 'rain'}
+            onClick={() => {
+              store.setState({ dataView: 'rain' });
```
Comment (ff6347) [inline src/components/Sidebar/SidebarSearch/index.tsx:51]: there should be an action for that

File: src/components/Sidebar/SidebarSearch/index.tsx [hunk 3]
```diff
@@ -1,12 +1,88 @@
...
+            }}
+          >
+            Niederschläge
+          </FilterLink>
+          <FilterLink
+            active={dataView === 'adopted'}
+            onClick={() => {
+              store.setState({ dataView: 'adopted' });
```
Comment (ff6347) [inline src/components/Sidebar/SidebarSearch/index.tsx:59]: there should be an action for that

File: src/components/Sidebar/SidebarSearch/index.tsx [hunk 4]
```diff
@@ -1,12 +1,88 @@
...
+            }}
+          >
+            Bereits adoptiert
+          </FilterLink>
+          <FilterLink
+            active={dataView === 'watered'}
+            onClick={() => {
+              store.setState({ dataView: 'watered' });
```
Comment (ff6347) [inline src/components/Sidebar/SidebarSearch/index.tsx:67]: there should be an action for that

File: src/components/UsersWateringsList/index.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+}> = ({ waterings }) => {
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const surpassedMaxItems = waterings.length > MAX_ITEMS;
+  const listItems = isExpanded ? waterings : waterings.slice(0, MAX_ITEMS);
+
+  return (
+    <WrapperOuter>
+      {listItems.map(
+        ({ username, timestamp, amount }: WateringType, i: number) => (
+          <Wrapper key={`Lastadopted-key-${i}`}>
```
Comment (ff6347) [inline src/components/UsersWateringsList/index.tsx:87]: The index might change. We should not use `i`

File: src/components/WaterNeedsInfo/index.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  display: block;
+  padding-top: 4px;
+`;
+
+const TreeWatering: FC = () => {
+  return (
+    <StyledWrapper>
+      {content.sidebar.waterNeeds.map(({ title, description }, i) => {
+        return (
+          <div key={`Tree-watering-${i}`}>
```
Comment (ff6347) [inline src/components/WaterNeedsInfo/index.tsx:36]: same here. Don't use `i`

File: src/utils/getWaterNeedByAge.ts [hunk 7]
```diff
@@ -1,16 +1,7 @@
...
-  }
-  if (age >= 15 && age < 40) {
-    return [1, 1];
-  }
-  if (age >= 40) {
-    return [1];
-  }
-  // TODO: How can this state happen?
-  return null;
+export const getWaterNeedByAge = (age?: number): number => {
```
Comment (ff6347) [inline src/utils/getWaterNeedByAge.ts:1]: this breaks a test

**Review by ff6347 — APPROVED** [review]
LGTM. The remarks are leftovers not newly introduced bugs. I think we also need a preview deploy, since seeing errors does sometimes not work in the code.

File: src/components/Sidebar/SidebarSearch/index.tsx (hunk 1 above)
Comment (vogelino) [reply src/components/Sidebar/SidebarSearch/index.tsx:19]: Will do!

File: src/components/UsersWateringsList/index.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    box-shadow: 0 0 0 2px ${p => p.theme.colorTextLight};
+  }
+`;
+
+const MAX_ITEMS = 8;
+
+const UsersWateringsList: FC<{
+  waterings: WateringType[];
+}> = ({ waterings }) => {
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
```
Comment (ff6347) [inline src/components/UsersWateringsList/index.tsx:79]: Type will be inferred from initial state. No need for `<boolean>` here

**Review by ff6347 — APPROVED** [review]
LGTM

File: src/components/UsersWateringsList/index.tsx (hunk 8 above)
Comment (vogelino) [reply src/components/UsersWateringsList/index.tsx:79]: Explicitness is always better. This forces the next dev to think about it when changing logic.

**Pull request merged**